web-animations-1/Overview.bs

```diff
@@ -3682,10 +3706,129 @@ as follows:
+
+<h4 id="removing-replaced-animations">Removing replaced animations</h4>
+
+An [=animation=] is
+<dfn lt="replaceable animation" local-lt="replaceable">replaceable</dfn>
+if <em>all</em> of the following conditions are true:
+
+*   The existence of the [=animation=] is <em>not</em> prescribed by markup.
+    That is, it is <em>not</em>
+    a CSS animation with an [=owning element (animation)|owning element=],
+    nor a CSS transition with an [=owning element (transition)|owning element=].
+*   The [=animation=]'s [=play state=] is [=finished play state|finished=].
+*   The [=animation=]'s [=replace state=] is <em>not</em>
+    [=removed replace state|removed=].
+*   The [=animation=] is associated with a [=monotonically increasing=]
+    [=timeline=].
+*   The [=animation=] has an associated [=target effect=].
+*   The [=target effect=] associated with the [=animation=] is [=in effect=].
+*   The [=target effect=] has an associated [=target element=].
+
+When asked to <dfn>remove replaced animations</dfn>
+for a {{Document}}, |doc|,
+then for every [=animation=], |animation|, that:
+
+*   has an [=associated with an animation|associated=] [=animation effect=]
+    whose [=target element=] is a [=descendant=] of |doc|, and
+
+*   is [=replaceable=], and
+
+*   has a [=replace state=] of [=active replace state|active=], and
+
+*   for which there exists for each [=target property=]
+    of every [=animation effect=]
+    [=associated with an animation|associated=] with |animation|,
+    an [=animation effect=] associated with a [=replaceable=] [=animation=]
+    with a higher [=composite order=] than |animation|
+    that includes the same [=target property=]
+
+perform the following steps:
+
+1.   Set |animation|'s [=replace state=] to [=removed replace state|removed=].
+
+1.   <a lt="create an event">Create</a> an {{AnimationPlaybackEvent}},
+     |removeEvent|.
+
+1.   Set |removeEvent|'s {{Event/type}} attribute to
+     <a lt="remove event">remove</a>.
+
+1.   Set |removeEvent|'s {{AnimationPlaybackEvent/currentTime}} attribute
+     to the [=current time=] of |animation|.
+
+1.   Set |removeEvent|'s {{AnimationPlaybackEvent/timelineTime}}
+     attribute to the <a lt="timeline current time">current time</a>
+     of the [=timeline=] with which |animation| is associated.
+
+1.   If |animation| has a [=document for timing=], then append
+     |removeEvent| to its [=document for timing=]'s
+     [=pending animation event queue=] along with its target, |animation|.
+     For the [=scheduled event time=], use the result of
+     applying the procedure to convert
+     [=timeline time to origin-relative time=] to
+     the [=timeline current time|current time=] of the [=timeline=]
+     with which |animation| is associated.
+
+     Otherwise, [=queue a task=] to [=dispatch=] |removeEvent| at
+     |animation|.
+     The task source for this task is the [=DOM manipulation task
+     source=].
+
```

Comment:
Should the animation then be removed? Is the replace state change sufficient to imply the animations is removed?

Reply:
Yes, setting the replace state to removed at the beginning of the procedure is sufficient for it to stop being returned from `getAnimations()` and stop contributing to the effect stack.